src/ucp/wireup/wireup_cm.c

```diff
@@ -438,21 +463,50 @@ static unsigned ucp_cm_client_uct_connect_progress(void *arg)
     /* To silence Coverity warning */
     UCS_STATIC_ASSERT(ucs_static_array_size(ep_init_flags_prio) > 0);
 
-    ucp_wireup_eps_pending_extract(ep, &tmp_pending_queue);
     for (i = 0; i < ucs_static_array_size(ep_init_flags_prio); ++i) {
         ep_init_flags = ep_init_flags_prio[i];
+        status        = ucp_cm_ep_client_initial_config_get(
+                ep, ep_init_flags, &cm_wireup_ep->cm_resolve_tl_bitmap, &key);
+        if (status != UCS_OK) {
+            goto try_fallback;
+        }
 
-        /* Cleanup the previously created UCP EP. The one that was created on
-         * the previous call to this client's resolve_cb */
-        ucp_wireup_cm_ep_cleanup(ep);
+        if (i < (ucs_static_array_size(ep_init_flags_prio) - 1)) {
+            /* Can use another variant of endpoint's initialization flags for
+             * fallback */
+            fallback_log_level = UCS_LOG_LEVEL_DEBUG;
+        } else if (ucp_cm_client_get_next_cm_idx(ep) != UCP_NULL_RESOURCE) {
+            /* Can use another CM for fallback */
+            fallback_log_level = UCS_LOG_LEVEL_DIAG;
+        } else {
+            /* No options for fallback */
+            fallback_log_level = UCS_LOG_LEVEL_ERROR;
+        }
```

Comment:
minor: move this to line 495, right before ucp_wireup_cm_priv_space_check

Reply:
Done